trade-finance-manager-ui/server/controllers/utilisation-reports/index.ts

```diff
@@ -1,30 +1,122 @@
 import { Request, Response } from 'express';
+import { subMonths } from 'date-fns';
 import api from '../../api';
 import { getFormattedReportDueDate, getFormattedReportPeriod } from '../../services/utilisation-report-service';
 import { getIsoMonth } from '../../helpers/date';
 import { getReportReconciliationSummaryViewModel } from './helpers';
 import { asString } from '../../helpers/validation';
+import { TfmSessionUser } from '../../types/tfm-session-user';
+import { ReportIdentifier, ReportWithStatus } from '../../types/utilisation-reports';
+import { UTILISATION_REPORT_RECONCILIATION_STATUS } from '../../constants';
+
+/**
+ * Gets the data required to render the /utilisation-reports page
+ */
+const getUtilisationReportsSummaryData = async (userToken: unknown) => {
+  const currentDate = new Date();
+  const oneIndexedMonth = subMonths(currentDate, 1).getMonth() + 1;
+  const year = currentDate.getFullYear();
+
+  const submissionMonth = getIsoMonth(currentDate);
+  const reconciliationSummaryApiResponse = await api.getUtilisationReportsReconciliationSummary(submissionMonth, asString(userToken));
+
+  const reportReconciliationSummary = getReportReconciliationSummaryViewModel(reconciliationSummaryApiResponse);
+  const reportPeriod = getFormattedReportPeriod();
+  const reportDueDate = await getFormattedReportDueDate(userToken);
+
+  return {
+    reportReconciliationSummary,
+    reportPeriod,
+    reportDueDate,
+    month: oneIndexedMonth,
+    year,
+  };
+};
 
 export const getUtilisationReports = async (req: Request, res: Response) => {
   const { userToken, user } = req.session;
 
   try {
-    const submissionMonth = getIsoMonth(new Date());
-    const reconciliationSummaryApiResponse = await api.getUtilisationReportsReconciliationSummary(submissionMonth, asString(userToken));
-
-    const reportReconciliationSummary = getReportReconciliationSummaryViewModel(reconciliationSummaryApiResponse);
-    const reportPeriod = getFormattedReportPeriod();
-    const reportDueDate = await getFormattedReportDueDate(userToken);
+    const { reportReconciliationSummary, reportPeriod, reportDueDate, month, year } = await getUtilisationReportsSummaryData(userToken);
 
     return res.render('utilisation-reports/utilisation-reports.njk', {
       user,
       activePrimaryNavigation: 'utilisation reports',
       reportReconciliationSummary,
       reportPeriod,
       reportDueDate,
+      month,
+      year,
     });
   } catch (error) {
     console.error('Error rendering utilisation reports page', error);
     return res.render('_partials/problem-with-service.njk', { user });
   }
 };
+
+const getReportIdentifiersFromUrl = (body: object, month: number, year: number): ReportIdentifier[] =>
+  Object.keys(body).reduce((reportIdentifiers, key) => {
```

Comment:
Just because I anticipate confusion around this function, here is an explanation for why I did it.

The form posts the information from all the inputs in the form as a payload. This includes things like the `_csrfToken` token, but also includes the checkboxes that *have* been ticked (if a checkbox is unticked, it is not sent in the payload, so the value will always be `'on'`). I named the checkboxes `set-status--(bankId|reportId)-(id)` such that `req.body` may look like
```
{
  _csrfToken: abc123,
  set-status--bankId-123: 'on',
  set-status--reportId-abcdef1234: 'on',
}
```
Knowing that it looks like this, the only way to tell which reports the user wants to update is to look through the keys and extract those which match the checkbox format. I then need to check whether or not a bank id or report id is supplied, hence the two groups `idType` and `id` below